tests/test_templatetags.py

```diff
@@ -164,3 +170,99 @@ def test_get_admin_interface_inline_template(self):
             "admin/edit_inline/stacked.html"
         )
         self.assertEqual(headless_template, "admin/edit_inline/headerless_stacked.html")
+
+    def test_get_active_date_hierarchy_none(self):
+        changelist = Mock()
+        changelist.date_hierarchy = None
+
+        date_field = templatetags.get_admin_interface_active_date_hierarchy(changelist)
+
+        self.assertIsNone(date_field)
+
+    def test_get_active_date_hierarchy_inactive(self):
+        changelist = Mock()
+        changelist.date_hierarchy = "last_login"
+        changelist.get_filters_params.return_value = {}
+
+        date_field = templatetags.get_admin_interface_active_date_hierarchy(changelist)
+
+        self.assertIsNone(date_field)
+
+    def test_get_active_date_hierarchy_active(self):
+        changelist = Mock()
+        changelist.date_hierarchy = "last_login"
+        params = {"some_field": 2, "last_login__year": 2022}
+        changelist.get_filters_params.return_value = params
```

Comment:
So these tests aren’t great because they are quite coupled to the exact implementation.
They also don’t test the URLs that are generated, which is the most important part of the feature!

My first thought was to use `create_autospec(ChangeList)`, but because all attributes are set in `__init__` and not defined on the class, mock complained when I set `mock.date_hierarchy`.

`create_autospec` also works with an instance instead of a class, but creating a changelist instance requires a model admin, a request, etc. which seemed too heavy.

Reply:
I used a mixed approach to make mocks use the real ChangeList methods and check the created URLs!

Reply:
These tests are very good considering that this is not an essential feature, thank you very much!